resources/generic.go

```diff
@@ -458,25 +458,12 @@ func (me *Generic) Read(ctx context.Context, d *schema.ResourceData, m any) diag
 				ctx = context.WithValue(ctx, settings.ContextKeyStateConfig, tfConfig)
 			}
 			err = service.Get(ctx, d.Id(), sttngs)
-		} else {
-			if restError, ok := err.(rest.Error); ok {
-				if restError.Code == 404 {
-					d.SetId("")
-					return diag.Diagnostics{}
-				}
-			}
-			return diag.FromErr(err)
-		}
-	}
-	if err != nil {
-		if restError, ok := err.(rest.Error); ok {
-			if restError.Code == 404 {
-				d.SetId("")
-				return diag.Diagnostics{}
-			}
 		}
-		return diag.FromErr(err)
+
+		d.SetId("")
```

Comment:
We have several behavior changes in here. It's not just the `404 shows an error` fix.

1. `d.SetId("")` and `diag.FromErr` is now always called, even if the `service.Get(ctx, d.Id(), sttngs)` before succeeds (the one inside the `re-run with confighcl` check)
2. `setId` is also called if it's a non-404 status code (before it was only for 404). Not sure if this is good or bad. If there is a server outage (e.g., 503), would this mean that the resource is removed from the state file?
3. If you create a resource via Terraform and then delete it via the API (outside of Terraform), will the next `terraform apply` produce an error instead of suggesting to create it?